custom_components/pioneer_async/media_player.py

```diff
@@ -440,21 +448,45 @@ async def volume_down() -> None:
 
         await self.pioneer_command(volume_down)
 
+    async def async_media_play(self) -> None:
+        """Send play command."""
+
+        async def media_play() -> None:
+            await self.pioneer.media_control("play")
+
+        await self.pioneer_command(media_play)
+
+    async def async_media_pause(self) -> None:
+        """Send pause command."""
+
+        async def media_pause() -> None:
+            await self.pioneer.media_control("pause")
+
+        await self.pioneer_command(media_pause)
+
+    async def async_media_stop(self) -> None:
+        """Send stop command."""
+
+        async def media_stop() -> None:
+            await self.pioneer.media_control("stop")
+
+        await self.pioneer_command(media_stop)
+
     async def async_media_previous_track(self) -> None:
         """Send previous track command."""
 
-        async def tuner_previous_preset() -> None:
-            await self.pioneer.tuner_previous_preset()
+        async def media_previous_track() -> None:
```

Comment:
The `media_previous_track` and `media_next_track` methods need to execute the media control commands conditional on the currently active source supporting media controls, without removing the ability to change between tuner presets when the tuner is the active source.

Reply:
Okay, maybe I misunderstood what `PioneerAVR.media_control()` is supposed to do. I thought its purpose was to select the right command from `MEDIA_CONTROL_COMMANDS`, depending on what is currently active. So if for example `TUNER` is active, `PioneerAVR.media_control('previous')` maps to `decrease_tuner_preset` (the same command that `self.pioneer.tuner_previous_preset()` also executes), but when `IPOD` is active, `PioneerAVR.media_control('previous')` maps to `operation_ipod_previous` instead. Is this not how this is supposed to work? What else is the purpose of `PioneerAVR.media_control()` then?